src/connector/src/sink/mqtt.rs

```diff
@@ -153,22 +223,90 @@ impl MqttSinkWriter {
     pub async fn new(
         config: MqttConfig,
         schema: Schema,
-        pk_indices: Vec<usize>,
         format_desc: &SinkFormatDesc,
-        db_name: String,
-        sink_from_name: String,
         id: u64,
     ) -> Result<Self> {
-        let formatter = SinkFormatterImpl::new(
-            format_desc,
-            schema.clone(),
-            pk_indices.clone(),
-            db_name,
-            sink_from_name,
-            &config.common.topic,
-        )
-        .await?;
+        let mut path = vec![];
+        if let Some(field) = &config.topic_field {
+            let mut iter = field.split('.');
```

Comment:
This piece of code is not covered by test cases. Could you please add a test case for nested-structured data?